### PR TITLE
Fix CodeCoverage Default Configuration

### PR DESCRIPTION
**Pull request by mayankbansal018**
## Description
Default CodeCoverage Configuration Ignoring Async methods.

## Related issue
Fixes https://developercommunity.visualstudio.com/content/problem/242314/code-coverage-doesnt-show-async-methods.html

### CODE REVIEW

File: src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/ExcludeFromCodeCoverage.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.CodeCoverage
+{
+    using System;
+
+    public class ExcludeFromCoverage : Attribute
```
Comment (cltshivash) [inline src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/ExcludeFromCodeCoverage.cs]: Please change the name.

**Review by cltshivash — APPROVED** [review]

File: src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs [hunk 2]
```diff
@@ -155,11 +155,12 @@ internal class EnableCodeCoverageArgumentExecutor : IArgumentExecutor
 @"            </Functions>" + Environment.NewLine +
 @"            <Attributes>" + Environment.NewLine +
 @"              <Exclude>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.DebuggerHidden.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.DebuggerNonUserCode.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Runtime.CompilerServices.CompilerGenerated.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.CodeDom.Compiler.GeneratedCode.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage.*</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.Diagnostics.DebuggerHiddenAttribute$</Attribute>" + Environment.NewLine +
```
Comment (smadala) [inline src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs]: We have this duplicate config here and tracedatacollector. try to keep it in one place.

File: src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs [hunk 3]
```diff
@@ -155,11 +155,12 @@ internal class EnableCodeCoverageArgumentExecutor : IArgumentExecutor
...
+@"                <Attribute>^System.Diagnostics.DebuggerNonUserCodeAttribute$</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.Runtime.CompilerServices.CompilerGeneratedAttribute$</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.CodeDom.Compiler.GeneratedCodeAttribute$</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute$</Attribute>" + Environment.NewLine +
+@"                <Attribute>^Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode.*</Attribute>" + Environment.NewLine +
```
Comment (smadala) [inline src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs]: Should we keep `TestSDKAutoGeneratedCodeAttribute`?

File: src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform
+{
+    using System;
+
+    public class TestSDKAutoGeneratedCode : Attribute
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs]: Nit: Doc.

File: src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs [hunk 5]
```diff
@@ -155,11 +155,12 @@ internal class EnableCodeCoverageArgumentExecutor : IArgumentExecutor
 @"            </Functions>" + Environment.NewLine +
 @"            <Attributes>" + Environment.NewLine +
 @"              <Exclude>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.DebuggerHidden.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.DebuggerNonUserCode.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Runtime.CompilerServices.CompilerGenerated.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.CodeDom.Compiler.GeneratedCode.*</Attribute>" + Environment.NewLine +
-@"                <Attribute>^System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage.*</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.Diagnostics.DebuggerHiddenAttribute$</Attribute>" + Environment.NewLine +
+@"                <Attribute>^System.Diagnostics.DebuggerNonUserCodeAttribute$</Attribute>" + Environment.NewLine +
```
Comment (smadala) [inline src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs]: What is issue with System.Diagnostics.DebuggerNonUserCode.*? Is it looking for System.Diagnostics.DebuggerNonUserCodeXYZAttribute? What was the error?

File: src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs (hunk 4 above)
Comment (smadala) [reply src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs]: Can we have general name similar to earlier commits which can solve https://github.com/Microsoft/vstest/issues/1548 .

File: src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs]: Nit: Namespace Microsoft.VisualStudio.TestPlatform.CodeCoverage or Microsoft.VisualStudio.TestPlatform.ObjectModel.CodeCoverage ?

File: src/package/nuspec/Microsoft.NET.Test.Sdk.targets [hunk 7]
```diff
@@ -63,7 +63,7 @@
     <ItemGroup Condition="'$(Language)'=='C#'">
       <Line Include="// &lt;auto-generated&gt; This file has been auto generated. &lt;/auto-generated&gt; "/>
       <Line Include="using System%3b "/>
-      <Line Include="[System.CodeDom.Compiler.GeneratedCode(&quot;Microsoft.NET.Test.SDK&quot;, &quot;15.6.0&quot;)] "/>
+      <Line Include="[Microsoft.VisualStudio.TestPlatform.TestSDKAutoGeneratedCode] "/>
```
Comment (smadala) [inline src/package/nuspec/Microsoft.NET.Test.Sdk.targets]: Should we add this to VB too?

**Review by smadala — CHANGES_REQUESTED** [review]

File: src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs (hunk 2 above)
Comment (mayankbansal018) [reply src/vstest.console/Processors/EnableCodeCoverageArgumentProcessor.cs]: We should check this for 15.8, but I don't want to take this in svc for 15.7

File: src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs (hunk 4 above)
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/CodeCoverage/TestSDKAutoGeneratedCode.cs]: This class won't make sense here. We have could define this file in target files.

**Pull request merged**